### ISSUE TITLE
A11y: adjust colour contrast for dark theme

### ISSUE THREAD

**Issue by hugovk** [squidfunk/mkdocs-material#5599]
### Bug description

With the dark theme, pages such as:

* https://squidfunk.github.io/mkdocs-material/getting-started/
* https://blog.pypi.org/

don't have enough contrast for the links, making them hard to read and affecting accessibility.

The ratio doesn't meet the [WCAG guidelines](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html). 

### Related links

- https://github.com/squidfunk/mkdocs-material/issues/4006 - mentions "too low contrasts for example for the header links"
- https://github.com/squidfunk/mkdocs-material/issues/1640 - says "I’m currently working on dark mode in https://github.com/squidfunk/mkdocs-material/pull/1639, which will also include a dedicated variable for link colors as proposed in https://github.com/squidfunk/mkdocs-material/issues/1583."

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11715314/example.zip)

### Steps to reproduce

1. Open site in dark mode
2. Inspect page in Chrome, select Lighthouse tab
3. Select Accessibility category (can uncheck others to speed it up)
4. Click "Analyze page load"
5. Check results

Expected results:

Score: 100 and no contrast issues 

Actual results:

Several contrast issues found:

<details>
<summary>Details</summary>

<img width="717" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1324225/bd38922b-9a23-4e1d-9d7b-ea08e9eb639f">
<img width="717" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1324225/180e1b03-4008-448c-94c3-528c594d767f">

</details>

In my view, the top priority would be to fix the colour used for links (`var(--md-typeset-a-color)`) which especially problematic for pages like https://blog.pypi.org/

Suggestion:

Inspect the problematic text using Chrome:

<details>
<summary>Details</summary>

<img width="1196" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1324225/82c37cdb-7d96-44f6-86cd-b05782822a0b">

</details>

Click the colour square in the styles pane and expand the bit next to Contrast ratio:

<details>
<summary>Details</summary>

<img width="309" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1324225/6c89f139-448f-4f86-97db-9f1a8a70e840">

</details>

Click the box next to AA 4.5 to get a suggested colour:

<details>
<summary>Details</summary>

<img width="298" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1324225/a2b0d0d1-4078-46a7-bef2-4282a9fbb7c0">

</details>

This gives `#7299e1` which is a good replacement for the problematic `#6c91d5`.

Note: make sure not to change colours for light mode at the same time, or it may make the contrast worse for light mode.

### Browser

Chrome

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. I agree that we can improve link colors in the theme-provided colors.

However, for PyPI, the story is different. It seems that they customized the theme, and [changed the primary color](https://github.com/pypi/warehouse/blob/main/docs/blog/stylesheets/extra.css) to their CI colors, but didn't adjust some other colors to fit dark mode.

Thus, I would recommend to raise this problem to the maintainers of the PyPI blog.

**Comment by hugovk** [comment]
Aha, thanks for pointing that out, I've opened https://github.com/pypi/warehouse/pull/13972 to fix the main PyPI contrast problem.

**Comment by hugovk** [comment]
Thank you!

**Comment by squidfunk** [comment]
Released as part of [9.2.0b3](https://github.com/squidfunk/mkdocs-material/releases/tag/9.2.0b3).
